### PR TITLE
Added MES acquisition with tests and tutorial notebook

### PR DESCRIPTION
**Pull request by henrymoss**
Added another acquisition function. 

For testing, I have only changed test_acquisitions.py. Should I also have written some custom tests?

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: emukit/bayesian_optimization/acquisitions/__init__.py [hunk 1]
```diff
@@ -7,3 +7,4 @@
 from .probability_of_improvement import ProbabilityOfImprovement  # noqa: F401
 from .probability_of_feasibility import ProbabilityOfFeasibility  # noqa: F401
 from .entropy_search import EntropySearch  # noqa: F401
+from .max_value_entropy_search import MaxValueEntropySearch
```
Comment (apaleyes) [inline emukit/bayesian_optimization/acquisitions/__init__.py]:
```suggestion
from .max_value_entropy_search import MaxValueEntropySearch  # noqa: F401
```

File: emukit/bayesian_optimization/acquisitions/max_value_entropy_search.py [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+                 num_samples: int = 10, grid_size: int = 5000) -> None:
+
+        """
+        MES acquisition function approximates the distribution of the value at the global
+        minimum and tries to decrease its entropy. See this paper for more details:
+        Z. Wang, S. Jegelka
+        Max-value Entropy Search for Efficient Bayesian Optimization
+        ICML 2017
+
+        This part of the code is heavily based on
```
Comment (apaleyes) [inline emukit/bayesian_optimization/acquisitions/max_value_entropy_search.py]: What do you mean by "heavily based"? At the end of the day it's just an implementation of the paper above, not sure if it is worth mentioning GPflowOpt here

File: emukit/bayesian_optimization/acquisitions/max_value_entropy_search.py [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+        Max-value Entropy Search for Efficient Bayesian Optimization
+        ICML 2017
+
+        This part of the code is heavily based on 
+        https://github.com/GPflow/GPflowOpt/blob/master/gpflowopt/acquisition/mes.py
+
+        :param model: GP model to compute the distribution of the minimum dubbed pmin.
+        :param space: Domain space which we need for the sampling of the representer points
+        :param num_samples: integer determining how many samples to draw of the minimum (does not need to be large)
+        :grid_size: number of random locations in grid used to fit the gumbel distribution and approximately generate the samples of the minimum (recommend scaling with problem dimension, i.e. 10000*d)
```
Comment (apaleyes) [inline emukit/bayesian_optimization/acquisitions/max_value_entropy_search.py]:
```suggestion
        :param grid_size: number of random locations in grid used to fit the gumbel distribution and approximately generate the samples of the minimum (recommend scaling with problem dimension, i.e. 10000*d)
```

File: emukit/bayesian_optimization/acquisitions/max_value_entropy_search.py [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+
+        self.model = model
+        self.space = space
+        self.num_samples = num_samples
+        self.grid_size = grid_size
+
+        # Initialize parameters to lazily compute them once needed
+        self.mins = None
+    def update_parameters(self):
+        self._prepare_sample()
```
Comment (apaleyes) [inline emukit/bayesian_optimization/acquisitions/max_value_entropy_search.py]: I am not sure there is any point in having this private method. I would just put the whole thing into "update_parameters" and use it as needed. Is there any reason you did it this way?

File: emukit/bayesian_optimization/acquisitions/max_value_entropy_search.py [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+        #sample K length vector from unif([0,1])
+        #return K Y* samples 
+        self.mins= -np.log(-np.log(np.random.rand(self.num_samples))) * beta + alpha
+
+
+    def _required_parameters_initialized(self):
+        """
+        Checks if all required parameters are initialized.
+        """
+        return not (self.mins is None )
```
Comment (apaleyes) [inline emukit/bayesian_optimization/acquisitions/max_value_entropy_search.py]:
```suggestion
        return self.mins is not None
```

**Comment by apaleyes** [comment]
As for testing, yes i believe we may need additional tests for this one. There must be some advice in documentation (contribution guide i believe) about how to see the coverage report

File: emukit/bayesian_optimization/acquisitions/max_value_entropy_search.py (hunk 4 above)
Comment (henrymoss) [reply emukit/bayesian_optimization/acquisitions/max_value_entropy_search.py]: I was copying the format of your entropy search acquisition . I can combine if you want?

Comment (apaleyes) [reply emukit/bayesian_optimization/acquisitions/max_value_entropy_search.py]: Frankly i missed that in the original one. I believe combining is a good idea, if you agree, please go ahead with that

File: tests/emukit/test_acquisitions.py [hunk 6]
```diff
@@ -6,7 +6,8 @@
 from scipy.optimize import check_grad
 
 from bayesian_optimization.test_entropy_search import entropy_search_acquisition
-from emukit.bayesian_optimization.acquisitions import ExpectedImprovement, NegativeLowerConfidenceBound, EntropySearch, MaxValueEntropySearch
+from emukit.bayesian_optimization.acquisitions import ExpectedImprovement, NegativeLowerConfidenceBound, EntropySearch
+from emukit.bayesian_optimization.acquisition import MaxValueEntropySearch
```
Comment (apaleyes) [inline tests/emukit/test_acquisitions.py]: That's weird, why is it coming from a different package name here?

Comment (henrymoss) [reply tests/emukit/test_acquisitions.py]: No idea!
I had to split the line, as it was too long.

Comment (apaleyes) [reply tests/emukit/test_acquisitions.py]: Yeah, that's fine, it's just that first line has "acquisition_s_" and second "acquisition"

File: emukit/bayesian_optimization/acquisitions/max_value_entropy_search.py (hunk 4 above)
Comment (henrymoss) [reply emukit/bayesian_optimization/acquisitions/max_value_entropy_search.py]: Done

File: notebooks/EMUKIT-Max-Value-Entropy-Search-Example.ipynb [hunk 7]
```diff
@@ -0,0 +1,203 @@
+{
```
Comment (apaleyes) [inline notebooks/EMUKIT-Max-Value-Entropy-Search-Example.ipynb]: <p>Can you proofread this intro a bit? Just things like typos and such.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/amzn/emukit/pull/239/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

Comment (henrymoss) [reply notebooks/EMUKIT-Max-Value-Entropy-Search-Example.ipynb]: Sorted!

**Pull request merged**